File: frontend/src/app/service/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { EmployeesCollection } from '../model/collection/employees-collection';
import { EmployeeProjectData } from '../model/dto/employee-project-data';
import { Employee } from '../model/entity/employee';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {
  
  private apiUrl: string = environment.apiUrl;
  
  constructor(private http: HttpClient) {
    this.apiUrl = this.apiUrl + "/employees";
  }
  
  public findAll(): Observable<EmployeesCollection> {
    return this.http.get<EmployeesCollection>(this.apiUrl);
  }
  
  public findById(employeeId: number): Observable<Employee> {
    return this.http.get<Employee>(this.apiUrl + "/" + employeeId);
  }
  
  public save(employee: Employee): Observable<Employee> {
    return this.http.post<Employee>(this.apiUrl, employee);
  }
  
  public update(employee: Employee): Observable<Employee> {
    return this.http.put<Employee>(this.apiUrl, employee);
  }
  
  public deleteById(employeeId: number): Observable<any> {
    return this.http.delete<any>(this.apiUrl + "/" + employeeId);
  }
  
  public findByEmployeeId(employeeId: number): Observable<EmployeeProjectData[]> {
    return this.http.get<EmployeeProjectData[]>(this.apiUrl + "/display-projects/" + employeeId);
  }
  
  
  
}

File: src/main/java/com/pfa/pack/service/EmployeeService.java
package com.pfa.pack.service;

import java.util.List;

import com.pfa.pack.model.dto.EmployeeAssignedProjectDto;
import com.pfa.pack.model.dto.collection.DtoCollection;
import com.pfa.pack.model.entity.Employee;

public interface EmployeeService {
	
	DtoCollection<Employee> findAll();
	Employee findById(final Integer employeeId);
	Employee save(final Employee employee);
	Employee update(final Employee employee);
	void deleteById(final Integer employeeId);
	List<Employee> findByDepartmentId(final Integer departmentId);
	List<Employee> findByManagerId(final Integer managerId);
	List<EmployeeAssignedProjectDto> findByManagerIdAndProjectId(final Integer managerId, final Integer projectId);
	List<Employee> findAllManagers();
	Employee findByUsername(final String username);
	void deleteByUsername(final String username);
	
}
